From 85792d33398e1297f56735b817b991a356152603 Mon Sep 17 00:00:00 2001
From: Lars Kanis <lars@greiz-reinsdorf.de>
Date: Mon, 14 Apr 2025 20:35:14 +0200
Subject: [PATCH 1/2] Fix making FFI::Function shareable for Ractor

This was broken since ruby commit
  https://github.com/ruby/ruby/commit/d80f3a287c5c8d0404b6cb837db360cab320cde1

It shrinked shareability of Proc objects, so that we have to avoid a Proc here.
---
 spec/ffi/function_spec.rb | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/spec/ffi/function_spec.rb b/spec/ffi/function_spec.rb
index 77d09d887..6c1486511 100644
--- a/spec/ffi/function_spec.rb
+++ b/spec/ffi/function_spec.rb
@@ -48,12 +48,14 @@ module LibTest
     expect(LibTest.testFunctionAdd(10, 10, function_add)).to eq(20)
   end
 
-  def adder(a, b)
-    a + b
+  module TestShareable
+    def self.call(a, b)
+      a + b
+    end
   end
 
   it "can be made shareable for Ractor", :ractor do
-    add = FFI::Function.new(:int, [:int, :int], &method(:adder))
+    add = FFI::Function.new(:int, [:int, :int], TestShareable)
     Ractor.make_shareable(add)
 
     res = Ractor.new(add) do |add2|
